File: exercice2/java/src/test/java/fr/devoxx/devops/logs/spark/Spark1Test.java
package fr.devoxx.devops.logs.spark;

import fr.devoxx.devops.logs.SparkTest;
import org.junit.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

public class Spark1Test extends SparkTest {
    @Test
    public void test_spark() {
        Spark1 test = new Spark1();
        long result = test.process(rdd());
        assertThat(result, is(25L));
    }
}

File: exercice2/scala/src/test/scala/fr/devoxx/devops/logs/spark/Spark1Test.scala
package fr.devoxx.devops.logs.spark

import fr.devoxx.devops.logs.SharedSparkContext
import org.scalatest.{FunSuite, Matchers}

class Spark1Test extends FunSuite with SharedSparkContext with Matchers {

  test("Les liens cassés") {
    val result = Spark1(rdd).process
    result shouldBe 25
  }
}
